Dev

VincentHuet opened the pull request.

Commit:
modification des sources pour le mailer

Commit:
correction suivant demande de Tanguy

tanguygo commented on app/models/translation.rb:
```diff
@@ -19,17 +19,27 @@ class Translation < ActiveRecord::Base
+
+
+  def needed_update?(current_translator)
+    self.needed_update == 1 && self.locale_id == current_translator.locale_id
```
need_updated? => plus correct comme nom en fait ici :)

tanguygo replied:
Et pour moi tu mixes 2 trucs ici.

Tu check si la translation need une update et si le translator est dans cette locale là.

Hors le 2eme point est plus une histoire de permissions. Ce n'est pas parce que le translator n'est pas dans la langue là que la translation ne doit pas être updatée.

tanguygo commented on app/models/translation.rb:
```diff
@@ -19,17 +19,27 @@ class Translation < ActiveRecord::Base
+      if locale_id == 1
+        phrase.translations..where("name != ?", :en).update_all(:needed_update => 1)
+      end
+    end
```
Y a "..".

Ca peut pas marcher.

Btw je ferai une méthode statique primary_locale sur le modele Locale qui renverrait la primary_locale.

Et tu check si le name != en dans le modele translation alors que ça devrait être dans le locale. Bref ça peut pas marcher ici je crois...

Commit:
mise de place du remplissage automatique de la table : yml_source
plus corrections
début des modification concernant le changement automatique du flag "need_updated"

Commit:
- listing automatique de toutes les fichiers yml
- chargement et comptabilisation du nombre de clefs ajoutées lors du click sur "load in DB"
- mail to => vincent@djengo.net lors du click sur "load in DB"
- contrôle d'unicité des clefs

Commit:
modification du mailling : envoi de mails uniquement si nouvelle clefs entrée

Commit:
mise en place des notification par mail lorsque la version en est modifié

Commit:
correction pour le mailling => opérationel
